File: app/src/main/java/com/arr/angel/pertpratice/db/ConverterQuestion.java
package com.arr.angel.pertpratice.db;

import android.arch.persistence.room.TypeConverter;

import com.arr.angel.pertpratice.model.Question;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class ConverterQuestion {

    @TypeConverter
    public static List<Question> fromStringQuestion(String value) {
        Type listType = new TypeToken<ArrayList<Question>>() {}.getType();
        return new Gson().fromJson(value, listType);
    }

    @TypeConverter
    public static String fromArrayListQuestion(List<Question> list) {
        Gson gson = new Gson();
        String json = gson.toJson(list);
        return json;
    }
}

File: app/src/main/java/com/arr/angel/pertpratice/db/ConverterQuestion.kt
package com.arr.angel.pertpratice.db

import androidx.room.TypeConverter
import com.arr.angel.pertpratice.model.Question
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types

class ConverterQuestion {
    private val moshi = Moshi.Builder().build()
    private val questionList = Types.newParameterizedType(
            List::class.java, Question::class.java
    )

    private val adapter = moshi.adapter<List<Question>>(questionList)

    @TypeConverter
    fun questionsToString(questions: List<Question> ) : String = adapter.toJson(questions)

    @TypeConverter
    fun jsonToQuestionsList(json: String) : List<Question> = adapter.fromJson(json) ?: emptyList()
}